pkg/api/steve/supportconfigs/generator.go

```diff
@@ -0,0 +1,169 @@
+package supportconfigs
+
+import (
+	"archive/tar"
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/pkg/types/config"
+	"github.com/sirupsen/logrus"
+	authzv1 "k8s.io/api/authorization/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/endpoints/request"
+	authv1 "k8s.io/client-go/kubernetes/typed/authorization/v1"
+)
+
+const (
+	// Endpoint The endpoint that this URL is accessible at - used for routing and for SARs
+	Endpoint             = "/v1/generateSUSERancherSupportConfig"
+	cspAdapterNamespace  = "cattle-csp-system"
+	cspAdapterConfigmap  = "csp-config"
+	tarContentType       = "application/x-tar"
+	unauthenticatedError = "unauthenticaed"
```

Comment:
```suggestion
	unauthenticatedError = "unauthenticated"
```

Reply:
I've removed this to comply with [this comment](https://github.com/rancher/rancher/pull/37713#discussion_r874238725)